blockonomics-woocommerce.php

```diff
@@ -207,30 +181,83 @@ function get_callback_url()
 
     function show_options()
     {
-        ?>
+        if (isset($_GET['tab']) && $_GET['tab'] == "currencies" && isset($_GET['settings-updated']) ? $_GET['settings-updated'] : '' == 'true')
+        {
+            $blockonomics = new Blockonomics;
+            $setup_errors = $blockonomics->testSetup();
+            $btc_error = isset($setup_errors['btc']) ? $setup_errors['btc'] : 'false';
+            $bch_error = isset($setup_errors['bch']) ? $setup_errors['bch'] : 'false';
+            $withdraw_requested = $blockonomics->make_withdraw();
+        }
+        if (isset($_GET['settings-updated']) ? $_GET['settings-updated'] : '' == 'true'){
+            ?><script type="text/javascript">
+            let redirect = sessionStorage.getItem("redirect");
+            if(redirect == 'true'){
+                sessionStorage.clear();
+                const urlParams = new URLSearchParams(window.location.href);
+                const currentTab = urlParams.get('tab') ?? 'settings';
+                if (currentTab === 'settings'){
+                    window.location.href = "options-general.php?page=blockonomics_options&tab=currencies";
+                } else {
+                    window.location.href = "options-general.php?page=blockonomics_options&tab=settings";
+                }
+            }
+            </script><?php
+        }
 
+        ?>
         <script type="text/javascript">
             function gen_secret() {
                 document.generateSecretForm.submit();
             }
+            function check_form(tab) {
+                const urlParams = new URLSearchParams(window.location.href);
+                const currentTab = urlParams.get('tab') ?? 'settings';
+                if (currentTab == tab){
+                    return;
+                }
+                if (document.getElementById('blockonomics_form_updated').value == 'true' || document.getElementById('blockonomics_api_updated').value == 'true'){
+                    if (currentTab === 'settings'){
```

Comment:
1) One solution is to save the form, using vanilla javascript or ajax request, before redirecting the user
```
if (currentTab === 'settings'){
    if(validateBlockonomicsForm()){
        var xhr = new XMLHttpRequest();
        xhr.open("POST", "options.php"); 
        xhr.onload = function(event){ 
            window.location.href = "options-general.php?page=blockonomics_options&tab=currencies";
        }; 
        var formData = new FormData(document.myform); 
        xhr.send(formData);
    }
} else {
    var xhr = new XMLHttpRequest();
    xhr.open("POST", "options.php"); 
    xhr.onload = function(event){ 
        window.location.href = "options-general.php?page=blockonomics_options&tab=settings";
    }; 
    var formData = new FormData(document.myform); 
    xhr.send(formData);
}
```
2) We could also use a hidden form and submit it, the same as how generateSecret works. We would be required to POST all the form values and save them ourselves.
```
if (currentTab === 'settings'){
    if(validateBlockonomicsForm()){
        document.saveFormAndRedirect.submit();
    }
} else {
    document.saveFormAndRedirect.submit();
}
```
3) We can remove autosaving unsaved changes during _tab changes_, as this is the cause of the added complexity.